packages/maskbook/src/plugins/Wallet/formatter.ts

```diff
@@ -26,8 +26,9 @@ export function formatBalance(balance: BigNumber, decimals: number, significant:
 
     let fraction = balance.modulo(base).toString(10) // (balance % base).toString(10)
 
+    const fractionFloatLength = fraction.match(/(\.\d+)?$/)?.[0].length ?? 0
```

Comment:
You may think there should not be balance with float number, the minimum unit of a token is always 1. But actually the balance we pass into `formatBalance()` is not always directly from a token balance, but a calculating result of bigNumber. E.g calculate swap token by the ratio.

Reply:
so why not add it to the code as comment 👀

Reply:
https://mikemcl.github.io/bignumber.js/#toFr

Reply:
`toFraction()` is not suitable in our case.

Reply:
The `formatBalance` should only handle integers.

Reply:
OK, convert all floats to integers before pass to formatBalance.